patches/minecraft/net/minecraft/world/World.java.patch

```diff
@@ -972,6 +974,31 @@
 +        return count;
 +    }
 +
++    public void markTileEntitiesInChunkForRemoval(Chunk chunk)
++    {
++        this.tileEntitiesChunkToBeRemoved.add(chunk.func_76632_l());
++    }
++
++    private void removeTileEntitiesForRemovedChunks()
++    {
++        if (!this.tileEntitiesChunkToBeRemoved.isEmpty())
++        {
++            java.util.function.Predicate<TileEntity> isInChunk = (tileEntity) ->
++            {
++                net.minecraft.util.math.ChunkPos tileChunkPos = new net.minecraft.util.math.ChunkPos(tileEntity.func_174877_v());
++                boolean inChunk = this.tileEntitiesChunkToBeRemoved.contains(tileChunkPos);
++                if (inChunk)
++                {
++                    tileEntity.onChunkUnload();
```

Comment:
This is only called once here. The other place this method is called is not active.